hw/arm/apple-m1.c

```diff
@@ -455,44 +454,57 @@ static void apple_m1_register_types(void)
 
 type_init(apple_m1_register_types);
 
+/* 
+ * Helper method to insert a flat image from a file into memory with a given alignment
+ * Offset is updated after load to the next unused address
+ */
+static int load_image_aligned(const char *filename, hwaddr* offset, uint64_t alignment, uint64_t max_sz) {
```

Comment:
I think it would be better if the `load_` helpers returned the load address instead of the size, since the load address is useful to calculate entry points, and the size can be retrieved if needed by doing `offset - load_address`.
They can probably also abort execution instead of returning -1 on error; recovering from a bad image file does not seem useful to me.
EDIT: Maybe replace the `max_sz` parameter with the max load address, so the caller doesn’t have to compute the max size itself?

Reply:
So I started working on this change but I ended up finding it a lot more annoying to work with than the current API. Partly because the return format for things like `load_image_targphys` which the image loading helper wraps already use returning a size or negative for errors as their API so I lose a place to return an error condition. It's likely there's a better pattern but I think that's something worth putting off for now.